### PR TITLE
feat(plugin-stealth): Add new user-agent-language-platform plugin

### PR DESCRIPTION
**Pull request by Niek**
This plugin basically replaces the following evasions:
https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth/evasions/accept-language
https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth/evasions/navigator.languages
At the same time, it fixes https://github.com/berstend/puppeteer-extra/issues/91

Instead of changing request headers and the ``navigator.languages`` (``navigator.language`` was missing by the way) properties as in the evasions above, we use call ``setUserAgentOverride`` in the CDP Network domain directly which is a better way of achieving this.

In addition, this evasion allows setting the ``navigator.platform`` and ``navigator.vendor`` properties, which makes it a lot less easy to detect puppeteer.

### CODE REVIEW

**Comment by berstend** [comment]
@Niek Interesting, thanks a lot for this PR!

I wasn't aware that you could modify the `Accept-Language` request header using `Network.setUserAgentOverride`.

Would you mind adding a test file that does before/after tests with this plugin?
You could use e.g. this file as the basis: https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin-stealth/evasions/accept-language/index.test.js

I would do it myself but there's no good way in Github (that I'm aware of) to push commits to this PR (without you using the "Contributor" status). :)

Is there a specific reason to add `navigator.vendor` here as well, as opposed to making a specific plugin for it? I personally lean towards making the evasions as modular as possible and splitting them up if it makes sense.

The rest (userAgent, acceptLanguage, platform) is set by a single `Network.setUserAgentOverride` call, so it make sense to group it into a single plugin.

Thanks again!

**Comment by berstend** [comment]
#91 is still something I need to look into (so end users can use the `adblock` plugin while still being able to do their own request interception), but if this PR fixes the `accept-language` evasion than this would make fixing #91 less critical in my opinion.

**Comment by berstend** [comment]
Once you add a test file you can specifically run it like so (from the root of the `puppeteer-extra-plugin-stealth` folder):

```bash
yarn ava -v ./evasions/webgl.vendor/index.test.js
```

File: packages/puppeteer-extra-plugin-stealth/evasions/user-agent-language-platform/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+
+  async onPageCreated(page) {
+    this.debug('onPageCreated - Will set these user agent options', {
+      opts: this.opts
+    })
+
+    page._client.send('Network.setUserAgentOverride', {
+      userAgent: this.opts.userAgent || (await page.browser().userAgent()),
```
Comment (berstend) [inline packages/puppeteer-extra-plugin-stealth/evasions/user-agent-language-platform/index.js]: Could it make sense to add `.replace('HeadlessChrome/', 'Chrome/')` here as well?

Comment (Niek) [reply packages/puppeteer-extra-plugin-stealth/evasions/user-agent-language-platform/index.js]: Yes, good idea! Added now

**Comment by Niek** [comment]
@berstend Thank *you* for the amazing repo! I will add the tests ASAP to the PR.

I have removed the ``navigator.vendor`` part, I agree that would probably be better in a separate plugin.

**Comment by Niek** [comment]
I added the tests, let me know if you need other changes @berstend

**Comment by berstend** [comment]
@Niek awesome, thanks a lot! Will look at the PRs asap. :)

**Comment by berstend** [comment]
@Niek thanks again and apologies for the delay (I'd like to blame the holidays) 😄 

I've removed the `user-agent`, `accept-language` evasions which are made obsolete by this one and renamed it `user-agent-override` (to keep it more concise).

Will merge in some other fixes & PRs and do a release afterwards. :)

Thanks! 👍

**Comment by berstend** [comment]
Released in `puppeteer-extra-plugin-stealth@2.4.7`. :) Thanks again!

**Comment by Niek** [comment]
Awesome, thanks for the great work!

**Comment by pastuh** [comment]
What if i want to randomize User Agent?
Can i somehow turn off user-agent-override but still use StealthPlugin ?

**Pull request merged**